### PR TITLE
Update search index when legacy-download-stats are set.

### PR DESCRIPTION
**Pull request by mhilton**
As part of the legacy statistics support, update the search index
every time the legacy-download-stats value is changed.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/659/
Test PASSed.

**Comment by mhilton** [comment]
To QA:
1. Run a charmstore and do a search (http :8080/v4/search?include=stats)
2. Update one of the entries (http PUT ":8080/v4/trusty/mysql-12/meta/extra-info" "Content-Type: application/json" "legacy-download-stats:=10000")
3. Do another search (http :8080/v4/search?include=stats). The updated charm should have the correct value and be higher in the search output.

File: internal/charmstore/search.go [hunk 1]
```diff
@@ -93,6 +93,27 @@ func (s *Store) UpdateSearch(r *charm.Reference) error {
...
 
+// UpdateSearchFields updates the search record for the entity reference r
+// with the updated values in fields.
+func (s *Store) UpdateSearchFields(r *charm.Reference, fields map[string]interface{}) error {
+	if s.ES == nil || s.ES.Database == nil {
+		return nil
+	}
+	var needUpdate bool
+	for k := range fields {
+		if k == "extrainfo.legacy-download-stats" {
```
Comment (urosj) [inline internal/charmstore/search.go:105]: I'd add a note for the future to add any other fields here as well that would trigger ES update.

**Comment by urosj** [comment]
Did QA, worked well. Code review is next.

**Comment by jrwren** [comment]
:+1: QA instructions are great. Thank You.

**Comment by urosj** [comment]
:+1: Small note for the future.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/664/
Test PASSed.

**Comment by mhilton** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**